Source/SwiftLintFramework/Rules/Lint/PeriodSpacingRule.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+import IDEUtils
+import SourceKittenFramework
+
+struct PeriodSpacingRule: SourceKitFreeRule, ConfigurationProviderRule, OptInRule, SubstitutionCorrectableRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "period_spacing",
+        name: "Period Spacing",
+        description: "Periods should not be followed by more than one space.",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("let pi = 3.2"),
+            Example("let pi = Double.pi"),
+            Example("let pi = Double. pi"),
+            Example("let pi = Double.  pi"),
+            Example("// A. Single."),
+            Example("///   - code: Identifier of the error. Integer."),
+            Example("""
+            // value: Multiline.
+            //        Comment.
+            """),
+            Example("""
+            /**
+            Sentence ended in period.
+
+            - Sentence 2 new line characters after.
+            **/
+            """)
+        ],
+        triggeringExamples: [
+            Example("/* Only god knows why.  This symbol does nothing. */", testWrappingInComment: false),
+            Example("// Only god knows why.  This symbol does nothing.", testWrappingInComment: false),
+            Example("// Single. Double.  End.", testWrappingInComment: false),
+            Example("// Single. Double.  Triple.   End.", testWrappingInComment: false),
+            Example("// Triple.   Quad.    End.", testWrappingInComment: false),
+            Example("///   - code: Identifier of the error.  Integer.", testWrappingInComment: false)
+        ],
+        corrections: [
+            Example("/* Why. ↓ Symbol does nothing. */"): Example("/* Why. Symbol does nothing. */"),
+            Example("// Why. ↓ Symbol does nothing."): Example("// Why. Symbol does nothing."),
+            Example("// Single. Double. ↓ End."): Example("// Single. Double. End."),
+            Example("// Single. Double. ↓ Triple. ↓  End."): Example("// Single. Double. Triple. End."),
+            Example("// Triple. ↓  Quad. ↓   End."): Example("// Triple. Quad. End."),
+            Example("///   - code: Identifier. ↓ Integer."): Example("///   - code: Identifier. Integer.")
+        ]
+    )
+
+    func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        // Find all comment tokens in the file and regex search them for violations
+        file.syntaxClassifications
+            .compactMap { (classifiedRange: SyntaxClassifiedRange) -> [NSRange]? in
+                switch classifiedRange.kind {
+                case .blockComment, .docBlockComment, .lineComment, .docLineComment:
+                    break
```

Comment:
Control flow is a bit unusual here. Can we extract this into a function or an extension method on `SyntaxClassifiedRange` and use a `guard` or `if` statement here instead?